networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DefaultEth2Peer.java

```diff
@@ -356,14 +353,18 @@ public SafeFuture<Void> requestBlobSidecarsByRange(
                       spec,
                       this,
                       listener,
-                      maxBlobsPerBlock.get(),
+                      maxBlobsPerBlock,
                       kzg,
                       request.getStartSlot(),
                       request.getCount()));
             })
         .orElse(failWithUnsupportedMethodException("BlobSidecarsByRange"));
   }
 
+  private int calculateMaxBlobsPerBlock(final UInt64 endSlot) {
+    return spec.atSlot(endSlot).getConfig().toVersionDeneb().orElseThrow().getMaxBlobsPerBlock();
```

Comment:
IMU we should not get the Deneb one here but rather use the misc helpers to get the right one (Deneb or Electra)

```suggestion
    return spec.atSlot(endSlot).miscHelpers().getMaxBlobsPerBlock().orElseThrow();
```

Reply:
I'm trying a different approach, putting back the "cross-milestone" configuration back to specConfig.
Let's see if it is good or not